oauth2client/contrib/gce.py

```diff
@@ -16,60 +16,113 @@
 """
-
+import datetime
 import json
 import logging
 import warnings
 
 import httplib2
 from six.moves import http_client
-from six.moves import urllib
 
-from oauth2client._helpers import _from_bytes
-from oauth2client import util
 from oauth2client.client import HttpAccessTokenRefreshError
 from oauth2client.client import AssertionCredentials
-
+from oauth2client._helpers import _from_bytes
 
 __author__ = 'jcgregorio@google.com (Joe Gregorio)'
 
 logger = logging.getLogger(__name__)
 
 # URI Template for the endpoint that returns access_tokens.
-_METADATA_ROOT = ('http://metadata.google.internal/computeMetadata/v1/'
-                  'instance/service-accounts/default/')
-META = _METADATA_ROOT + 'token'
-_DEFAULT_EMAIL_METADATA = _METADATA_ROOT + 'email'
+_METADATA_ROOT = 'http://metadata.google.internal/computeMetadata/v1/'
 _SCOPES_WARNING = """\
 You have requested explicit scopes to be used with a GCE service account.
-Using this argument will have no effect on the actual scopes for tokens
-requested. These scopes are set at VM instance creation time and
-can't be overridden in the request.
+These scopes *are* present on the credentials; but this argument will have
+no effect on the actual scopes for tokens requested. The credentials scopes
+are set at VM instance creation time and can't be overridden in the request.
+To learn more go to: https://cloud.google.com/compute/docs/authentication
 """
+_SCOPES_ERROR = """\
+You have requested explicit scopes to be used with a GCE service account
+which are not available on the credentials. The scopes are set at VM instance
+creation time and can't be overridden in the request.
+To learn more go to: https://cloud.google.com/compute/docs/authentication
+"""
+_NOW = datetime.datetime.now
+
 
+class MetadataServerHttpError(Exception):
+    """Error for Http failures originating from the Metadata Server"""
 
-def _get_service_account_email(http_request=None):
-    """Get the GCE service account email from the current environment.
+
+def _get_metadata(http_request=None, path=None, recursive=True):
+    """Gets a JSON object from the specified path on the Metadata Server
 
     Args:
-        http_request: callable, (Optional) a callable that matches the method
-                      signature of httplib2.Http.request, used to make
-                      the request to the metadata service.
+        http_request: an httplib2.Http().request object or equivalent
+            with which to make the call to the metadata server
+        path: a list of strings denoting the metadata server request
+            path.
 
     Returns:
-        tuple, A pair where the first entry is an optional response (from a
-        failed request) and the second is service account email found (as
-        a string).
+        A deserialized JSON object representing the data returned
+        from the metadata server
     """
-    if http_request is None:
+
+    if not path:
```

Comment:
Do you actually mean to trigger on any falsey value, or should this be an explicit `is None`?